src/frontend/components/component-info/component-info.ts

```diff
@@ -32,68 +32,47 @@ export class ComponentInfo {
 
   private ComponentLoadState = ComponentLoadState;
 
-  private get path(): Path {
-    return deserializePath(this.node.id);
-  }
-
-  private get inputs(): InputOutput {
-    if (this.node == null || this.node.input == null) {
-      return {};
-    }
-
-    return this.node.input.reduce(
-      (accum, input) => {
-        const [name, alias] = input.split(/:/);
-        accum[name] = {alias};
-        return accum;
-      },
-      <InputOutput> {});
-  }
-
-  private get outputs(): InputOutput {
-    if (this.node == null) {
-      return {};
-    }
+  path: Path;
 
-    return this.node.output.reduce(
-      (accum, output) => {
-        const [name, alias] = output.split(/:/);
-        accum[name] = {alias};
-        return accum;
-      },
-      <InputOutput> {});
-  }
+  hasProviders: boolean = false;
+  hasDirectives: boolean = false;
+  hasDependencies: boolean = false;
+  hasProperties: boolean = false;
+  hasState: boolean = false;
 
-  private get hasState() {
-    if (this.node == null || this.state == null) {
-      return false;
-    }
+  inputs: InputOutput;
+  outputs: InputOutput;
 
-    return Object.keys(this.state).length > 0;
+  ngOnInit() {
+    this.ngOnChanges();
   }
 
-  private get hasProviders() {
-    return this.node &&
-        this.node.providers &&
-        this.node.providers.length > 0;
-  }
+  ngOnChanges() {
```

Comment:
@xorgy a lot of this stuff has been moved to the onChanges hook, which implies that components can have providers or other things added/removed at runtime.

I am almost certain that I am missing something, and probably out of the loop, but would like to understand the reasons behind this resetting everything everytime onChanges gets called.